### DISCUSSION TITLE
How should I get the latest package version using `NuGet.Client`

### DISCUSSION THREAD

**Discussion by WeihanLi** [NuGet/Home#13045] [Q&A]
Hi there, I'm trying to get the latest package version of a package

```c#
Task<NuGetVersion?> GetLatestPackageVersion(string packageId, 
    bool includePrerelease = false, CancellationToken cancellationToken = default)
```

Is there an API for this?

Found some ways as follows:

- `PackageMetadataResource.GetPackageMetadataAsync`
- `FindPackageByIdResource.GetAllVersionsAsync`

Are there more ways for it I missed? And which is recommended? Thanks very much.

**Comment by ghost** [comment]
Issue is missing Type label, remember to add a [Type label](https://github.com/NuGet/Client.Engineering/blob/main/designs/nuget-issues-approach.md#issue-type)

**Comment by nkolev92** [comment]
Those are the 2 ways, and the bottom line is depending on what information you want.

Those 2 methods return different data, one returns a list of versions, the other one includes metadata such as whether a package is listed, whether a package is vulnerable etc, so it really depends on your scenario.

**Reply by WeihanLi** [reply]
Thanks very much for the reply, I only need the latest package version, think `GetPackageMetadataAsync` may return too much for me, so I use the `GetAllVersionsAsync` for now, and is there a `GetLatestVersion(bool prerelease)` API or something like that

**Reply by nkolev92** [reply]
Another consideration is that `PackageMetadataResource` has means to ignore unlisted versions. FindPackageByIdResource will consider all versions listed or unlisted.
